### PR TITLE
Add default java images

### PR DESCRIPTION
**Pull request by wbobeirne**
Another simple one, just installs the highest JDK version that both CentOS and Ubuntu had in their repositories (14). We could go through the trouble of downloading 15 and manually installing, but this just seemed easiest.

### CODE REVIEW

**Review by vapurrmaid — COMMENTED** [review]
Not currently working too heavily with Java, but it might be nice to include:

- https://sdkman.io/

```bash
curl -s "https://get.sdkman.io" | bash
source "$HOME/.sdkman/bin/sdkman-init.sh"
```

_possibly_ include in a coder/configure script? :

- https://sdkman.io/sdks#gradle
- https://sdkman.io/sdks#maven

as I can't imagine doing most real projects without one or the other.

That being said, I wouldn't hold up without any of the above.

Thoughts?

File: images/java/Dockerfile.ubuntu [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM codercom/enterprise-base:ubuntu
+
+# Run everything as root
+USER root
+
+# Install JDK (OpenJDK 14)
+RUN DEBIAN_FRONTEND="noninteractive" apt-get install -y openjdk-14-jdk
```
Comment (jakelang) [inline images/java/Dockerfile.ubuntu:7]: Needs maven and gradle

File: images/java/Dockerfile.centos [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM codercom/enterprise-base:centos
```
Comment (jakelang) [inline images/java/Dockerfile.centos:1]: What's the rationale for using a CentOS base as opposed to Debian or other?

File: images/java/Dockerfile.ubuntu (hunk 1 above)
Comment (jakelang) [inline images/java/Dockerfile.ubuntu:7]: Also - it seems that OpenJDK 11 is the most commonly used Java version nowadays, after Java 8. In fact, IntelliJ IDEs come with a packaged OpenJDK JRE 11.

Comment (wbobeirne) [reply images/java/Dockerfile.ubuntu:7]: Ah yeah no kidding, looks like Java 8 is still ~65% of market share, I'll just switch it to use that.

**Comment by wbobeirne** [comment]
Added maven and gradle, avoided sdkman since it expects a lot to be done in user space in a way that I wasn't sure how it would work in Coder Enterprise (maybe needs a configure script or something?)

**Review by vapurrmaid — APPROVED** [review]
Thanks for getting maven and gradle in, it just felt like the right move to reduce inertia for most projects. 

I might have created a script for each of the installs and invoked that instead of duplicating script contents inline across different dockerfiles, but this is also fine :ok_hand: for these dockerfiles are manageable in the current state

**Pull request merged**